src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
+  async githubAuthCallback(@Query('code') code: string) {
+    try {
+      const jwtToken = await this.authService.handleGitHubCallback(code);
+
+      return jwtToken;
```

Comment:
PullProofAI

The `githubAuthCallback` method in the `AuthController` lacks proper error handling and response structure. It's recommended to handle errors consistently and return responses with appropriate status codes and messages.

```diff
+ try {
+     const jwtToken = await this.authService.handleGitHubCallback(code);
+
+     return jwtToken;
+ } catch (error) {
+     if (error.status) {
+         throw new HttpException(error.message, error.status);
+     } else {
+         throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+     }
+ }
```



<!-- This is an auto-generated comment by PullProofAI -->